common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/ISmartcardCertBasedAuthManager.java

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.ui.webview.challengehandlers;
+
+import androidx.annotation.NonNull;
+
+/**
+ * An abstract manager that can control connections for a particular type of smartcard.
+ */
+public interface ISmartcardCertBasedAuthManager {
+    /**
+     * Logic to prepare an Android device to detect smartcards via usb.
+     * @param startDiscoveryCallback Contains callbacks to run when a smartcard is connected and disconnected.
+     */
+    void startDiscovery(final IStartDiscoveryCallback startDiscoveryCallback);
+
+    /**
+     * Cease usb discovery of smartcards.
+     */
+    void stopDiscovery();
+
+    /**
+     * Attempt to get an instance of a session in order to carry out methods specific to ISmartcardSession.
+     * @param callback Contains callbacks to run when a ISmartcardSession is successfully instantiated and when any exception is thrown due to a connection issue.
+     */
+    void attemptDeviceSession(@NonNull final ISessionCallback callback);
+
+    /**
+     * Returns boolean based on if a smartcard device is currently connected to the Android device and detected by our code.
+     * @return true if a device is currently connected, false otherwise.
+     */
+    boolean isDeviceConnected();
+
+    /**
+     * Runs smartcard brand specific processes that may need to occur prior to authentication.
+     */
+    void prepareForAuth();
+
+    /**
+     * Callback methods to be run upon initial connection and disconnection of a smartcard device.
+     */
+    interface IStartDiscoveryCallback {
```

Comment:
Can this ever error out? Should there be an error handling callback here too?

(Even if Yubikey APIs won't throw an error, what about another OEM?)